tools/generate-wire.py

```diff
@@ -410,8 +410,12 @@ def print_towire(self,is_header):
         #    commit_sig,0,channel-id,u64
         for m in messages:
             if m.name == parts[0]:
-                m.addField(Field(parts[0], parts[2], parts[3], comments, prevfield))
-                prevfield = parts[2]
+                f = Field(parts[0], parts[2], parts[3], comments, prevfield)
```

Comment:
Is there a reason we don't remember all previous fields, instead of just the previous field? Seems like keeping a `dict` that allows us to reference any previous fields is quite easy.